keras/regularizers.py

```diff
@@ -1,26 +1,117 @@
 from __future__ import absolute_import
-import theano
 import theano.tensor as T
-import numpy as np
-
-def l1(l=.01):
-    def l1wrap(g, p):
-        g += T.sgn(p) * l
-        return g
-    return l1wrap
-
-def l2(l=.01):
-    def l2wrap(g, p):
-        g += p * l
-        return g
-    return l2wrap
-
-def l1l2(l1=.01, l2=.01):
-    def l1l2wrap(g, p):
-        g += T.sgn(p) * l1
-        g += p * l2
-        return g
-    return l1l2wrap
-
-def identity(g, p):
-    return g
\ No newline at end of file
+
+
+class Regularizer(object):
+    def __init__(self):
+        self.modifies_gradient = False
+        self.modifies_cost = False
+
+    def update_gradient(self, gradient, params):
+        raise NotImplementedError
+
+    def update_cost(self, cost):
+        raise NotImplementedError
+
+
+class RegularizerWeightsL1(Regularizer):
```

Comment:
What about simply `L1`? It lives inside `regularizers` and inherits from `Regularizer`, so it's pretty clear it's a regularizer.